apps/studio/src/components/PageEditor/MenuBar/AccordionMenuBar.tsx

```diff
@@ -82,9 +92,49 @@ export const AccordionMenuBar = ({ editor }: { editor: Editor }) => {
           },
         ],
       },
+      {
+        type: "divider",
+      },
+      {
+        type: "item",
+        icon: BiLink,
+        title: "Link",
+        action: onLinkModalOpen,
+        isActive: () => editor.isActive("link"),
+      },
     ],
     [editor],
   )
 
-  return <MenuBar items={items} />
+  return (
+    <>
+      <LinkEditorModal
+        linkText={
+          editor.isActive("link")
+            ? editor.state.doc.nodeAt(
+                Math.max(1, editor.view.state.selection.from - 1),
+              )?.textContent
+            : ""
+        }
+        linkHref={
+          editor.isActive("link") ? editor.getAttributes("link").href : ""
+        }
+        onSave={(linkText, linkHref) => {
+          editor
+            .chain()
+            .focus()
+            .extendMarkRange("link")
+            .deleteSelection()
+            .insertContent(`<a href="${linkHref}">${linkText}</a>`)
+            .run()
+
+          onLinkModalClose()
+        }}
+        onClose={onLinkModalClose}
+        isOpen={isLinkModalOpen}
+      />
```

Comment:
seems like repeated code, might miss one when we change the other next time. Should we extract it out to a reusable hook instead?

```ts
const useLinkEditorModal = (editor) => {
  // setup
  // ...

  return {
    linkText,
    linkHref,
    onSave,
  }
}
```

Reply:
Switch to the hook method but returns `onLinkModalOpen` and `LinkEditorModal` instead for consumption.